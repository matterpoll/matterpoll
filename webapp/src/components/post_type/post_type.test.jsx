import React from 'react';
import {shallow} from 'enzyme';

import PostType from 'components/post_type/post_type';

describe('components/post_type/PostType', () => {
    const baseProps = {
        post: {
            props: {
                attachments: [
                    {
                        author_name: 'sample_name',
                        title: 'sample_title',
                        text: 'sample_text',
                    },
                ],
            },
        },
        theme: {},
        options: {},
        postTypeComponentId: '',
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<PostType {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot without any attachments', () => {
        const newProps = {
            ...baseProps,
        };
        newProps.post.props.attachments = [];

        const wrapper = shallow(<PostType {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with two attachments', () => {
        const newProps = {
            ...baseProps,
        };
        newProps.post.props.attachments.push({
            author_name: 'sample_name',
            title: 'sample_title',
            text: 'sample_text',
        });

        const wrapper = shallow(<PostType {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
