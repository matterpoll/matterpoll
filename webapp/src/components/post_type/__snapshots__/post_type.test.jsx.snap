// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_type/PostType should match snapshot 1`] = `
<div
  className="attachment"
>
  <div
    className="attachment__content"
  >
    <div
      className="clearfix attachment__container"
    >
      <span
        className="attachment__author-name"
        key="attachment__author-name"
      >
        sample_name
      </span>
      <h1
        className="attachment__title"
      >
        mockMessageHtmlToComponent(mockFormatText(sample_title))
      </h1>
      <div>
        <div
          className="attachment__body attachment__body--no_thumb"
        >
          mockMessageHtmlToComponent(mockFormatText(sample_text))
          <FieldsTable
            attachment={
              Object {
                "author_name": "sample_name",
                "text": "sample_text",
                "title": "sample_title",
              }
            }
            options={Object {}}
          />
          <Connect(ActionView)
            attachment={
              Object {
                "author_name": "sample_name",
                "text": "sample_text",
                "title": "sample_title",
              }
            }
            post={
              Object {
                "props": Object {
                  "attachments": Array [
                    Object {
                      "author_name": "sample_name",
                      "text": "sample_text",
                      "title": "sample_title",
                    },
                  ],
                },
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "clear": "both",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`components/post_type/PostType should match snapshot with two attachments 1`] = `
<div
  className="attachment"
>
  <div
    className="attachment__content"
  >
    <div
      className="clearfix attachment__container"
    >
      <span
        className="attachment__author-name"
        key="attachment__author-name"
      >
        sample_name
      </span>
      <h1
        className="attachment__title"
      >
        mockMessageHtmlToComponent(mockFormatText(sample_title))
      </h1>
      <div>
        <div
          className="attachment__body attachment__body--no_thumb"
        >
          mockMessageHtmlToComponent(mockFormatText(sample_text))
          <FieldsTable
            attachment={
              Object {
                "author_name": "sample_name",
                "text": "sample_text",
                "title": "sample_title",
              }
            }
            options={Object {}}
          />
          <Connect(ActionView)
            attachment={
              Object {
                "author_name": "sample_name",
                "text": "sample_text",
                "title": "sample_title",
              }
            }
            post={
              Object {
                "props": Object {
                  "attachments": Array [
                    Object {
                      "author_name": "sample_name",
                      "text": "sample_text",
                      "title": "sample_title",
                    },
                  ],
                },
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "clear": "both",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`components/post_type/PostType should match snapshot without any attachments 1`] = `
<div
  className="attachment"
>
  <div
    className="attachment__content"
  >
    <div
      className="clearfix attachment__container"
    >
      <div>
        <div
          className="attachment__body attachment__body--no_thumb"
        >
          <FieldsTable
            attachment={Object {}}
            options={Object {}}
          />
          <Connect(ActionView)
            attachment={Object {}}
            post={
              Object {
                "props": Object {
                  "attachments": Array [],
                },
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "clear": "both",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
